modules/common/condition/funcs_test.go

```diff
@@ -165,6 +165,55 @@ func TestSet(t *testing.T) {
 	g.Expect(c2.LastTransitionTime).To(BeIdenticalTo(time1))
 }
 
+func TestRemove(t *testing.T) {
+	conditions := Conditions{}
+
+	tests := []struct {
+		name      string
+		condition *Condition
+		want      Conditions
+	}{
+		{
+			name:      "Add nil condition",
+			condition: nil,
+			want:      CreateList(unknownReady),
+		},
+		{
+			name:      "Add unknownB condition",
+			condition: unknownB,
+			want:      CreateList(unknownReady, unknownB),
+		},
+		{
+			name:      "Add another condition unknownA, gets sorted",
+			condition: unknownA,
+			want:      CreateList(unknownReady, unknownA, unknownB),
+		},
+		{
+			name:      "Add same condition unknownA, won't duplicate",
+			condition: unknownA,
+			want:      CreateList(unknownReady, unknownA, unknownB),
+		},
+	}
+
+	g := NewWithT(t)
+
+	conditions.Init(nil)
+	g.Expect(conditions).To(haveSameConditionsOf(CreateList(unknownReady)))
```

Comment:
it is already tested above that an empty conditions init will result in a single Ready condition